openslides/core/static/js/core/site.js

```diff
@@ -525,33 +525,88 @@ angular.module('OpenSlidesApp.core.site', [
 .controller('SearchBarCtrl', [
     '$scope',
     '$state',
-    function ($scope, $state) {
-        $scope.search = function(query) {
-            $scope.query = '';
+    '$sanitize',
+    function ($scope, $state, $sanitize) {
+        $scope.search = function() {
+            var query = $sanitize($scope.querybar);
+            $scope.querybar = '';
             $state.go('search', {q: query});
         };
```

Comment:
upon calling this `search()`, we go to a different $scope, where `$scope.query` is undefined yet. So we need to pass the content of `$scope.querybar` to the *other* $scope. That's when `q` is used.
@normanjaeckel